File: javascript/node.spec.js
var Node = require('./node');

describe('Node', function() {

  it('does not have a link by default', function() {
    var node = new Node('hello');
    expect(node.link).toBeUndefined();
  });

  it('can have a link set', function() {
    var node = new Node('hello');
    node.link = new Node('node');
    console.log(node.link)
    expect(node.link.data).toEqual('node')
  });

});

File: linked_list/node_spec.rb
require './node'

describe Node do

  it "exists" do
    expect(Node).to be
  end

  it "does not have a link by default" do
    node = Node.new
    expect(node.link).to be(nil)
  end

  it "can be linked to another node" do
    node1 = Node.new
    node2 = Node.new
    node1.link = node2
    expect(node1.link).to be(node2)
  end

end
